### PR TITLE
Tests for api.py 

### PR DESCRIPTION
**Pull request by cudkowicz**
Added tests to api for both possibilities of the method offer_get, contract_get, and offer_update.

### CODE REVIEW

File: esi_leap/tests/db/sqlalchemy/test_api.py [hunk 1]
```diff
@@ -259,4 +266,122 @@ def test_offer_get_by_name(self):
 
     def test_offer_get_by_name_not_found(self):
 
-        assert api.offer_get_by_uuid('some_uuid') is None
+        self.assertEqual(api.offer_get_by_name('some_name'), [])
+
+    def test_offer_destroy(self):
+
```
Comment (tzumainn) [inline esi_leap/tests/db/sqlalchemy/test_api.py:272]: You don't actually need this blank line after each function definition (I think you probably saw one early on and thought it was the standard).

File: esi_leap/tests/db/sqlalchemy/test_api.py [hunk 2]
```diff
@@ -259,4 +266,122 @@ def test_offer_get_by_name(self):
...
+        test_contract_1['offer_uuid'] = o1.uuid
+        test_contract_2['offer_uuid'] = o2.uuid
+        test_contract_3['offer_uuid'] = o3.uuid
+        test_contract_4['offer_uuid'] = o4.uuid
+        c1 = api.contract_create(test_contract_1)
+        c2 = api.contract_create(test_contract_2)
+        c3 = api.contract_create(test_contract_3)
+        api.contract_create(test_contract_4)
+        res = api.contract_get_by_name('c1')
+        assert len(res) == 3
```
Comment (tzumainn) [inline esi_leap/tests/db/sqlalchemy/test_api.py:327]: I was confused why you created so many contracts for this, but you're right, this function can return multiple results. Good thinking!

File: esi_leap/tests/db/sqlalchemy/test_api.py [hunk 3]
```diff
@@ -259,4 +266,122 @@ def test_offer_get_by_name(self):
...
+        self.assertEqual(api.offer_get_by_uuid('offer_2'), None)
+
+    def test_offer_destroy_not_found(self):
+
+        self.assertEqual(api.offer_get_by_uuid('offer_4'), None)
+
+    def test_offer_update(self):
+
+        offer = api.offer_create(test_offer_3)
+        o = api.offer_create(test_offer_2)
```
Comment (tzumainn) [inline esi_leap/tests/db/sqlalchemy/test_api.py:284]: You don't need to actually create this offer; you can just use test_offer_2['start_time'] below.

File: esi_leap/tests/db/sqlalchemy/test_api.py [hunk 4]
```diff
@@ -259,4 +266,122 @@ def test_offer_get_by_name(self):
...
+                  'end_time': o.end_time}
+        api.offer_update(offer.uuid, values)
+        res = api.offer_get_by_uuid(offer.uuid)
+        self.assertEqual(res.start_time, offer.start_time)
+        self.assertEqual(res.end_time, offer.end_time)
+
+    def test_offer_get_all(self):
+
+        o = api.offer_create(test_offer_2)
+        offer = api.offer_create(test_offer_3)
```
Comment (tzumainn) [inline esi_leap/tests/db/sqlalchemy/test_api.py:295]: I would use variable names like `o1` and `o2`; if the objects are similar, use similar variable names

File: esi_leap/tests/db/sqlalchemy/test_api.py [hunk 5]
```diff
@@ -259,4 +266,122 @@ def test_offer_get_by_name(self):
...
+        res = api.offer_get_all({}).all()
```
Comment (tzumainn) [inline esi_leap/tests/db/sqlalchemy/test_api.py:296]: Do you need the `all()` here?

File: esi_leap/tests/db/sqlalchemy/test_api.py [hunk 6]
```diff
@@ -259,4 +266,122 @@ def test_offer_get_by_name(self):
...
+        self.assertEqual((o.to_dict(), offer.to_dict()),
+                         (res[0].to_dict(), res[1].to_dict()))
+
+    def test_contract_get_by_uuid(self):
+
+        o = api.offer_create(test_offer_2)
+        test_contract_4['offer_uuid'] = o.uuid
```
Comment (tzumainn) [inline esi_leap/tests/db/sqlalchemy/test_api.py:303]: Ah, don't do this; in the test_contract_4 dict, hardcode the offer_uuid key to the offer uuid (likely test_offer_4's uuid).

File: esi_leap/tests/db/sqlalchemy/test_api.py (hunk 1 above)
Comment (cudkowicz) [reply esi_leap/tests/db/sqlalchemy/test_api.py:272]: I deleted all the blank lines after the functions defs.

File: esi_leap/tests/db/sqlalchemy/test_api.py (hunk 3 above)
Comment (cudkowicz) [reply esi_leap/tests/db/sqlalchemy/test_api.py:284]: ok ill change that around.

File: esi_leap/tests/db/sqlalchemy/test_api.py (hunk 4 above)
Comment (cudkowicz) [reply esi_leap/tests/db/sqlalchemy/test_api.py:295]: makes sense

File: esi_leap/tests/db/sqlalchemy/test_api.py (hunk 3 above)
Comment (cudkowicz) [reply esi_leap/tests/db/sqlalchemy/test_api.py:284]: i needed to create the offer in order to create the res to check the update against

File: esi_leap/tests/db/sqlalchemy/test_api.py (hunk 5 above)
Comment (cudkowicz) [reply esi_leap/tests/db/sqlalchemy/test_api.py:296]: nope!

File: esi_leap/tests/db/sqlalchemy/test_api.py (hunk 6 above)
Comment (cudkowicz) [reply esi_leap/tests/db/sqlalchemy/test_api.py:303]: ok ill fix that up!

File: esi_leap/tests/db/sqlalchemy/test_api.py (hunk 3 above)
Comment (tzumainn) [reply esi_leap/tests/db/sqlalchemy/test_api.py:284]: You need to create test_offer_3, but you don't need to create test_offer_2, right?

Comment (cudkowicz) [reply esi_leap/tests/db/sqlalchemy/test_api.py:284]: oh i see now.

File: esi_leap/tests/db/sqlalchemy/test_api.py (hunk 6 above)
Comment (cudkowicz) [reply esi_leap/tests/db/sqlalchemy/test_api.py:303]: i remember having an issue with this working when we did it in the past. The problem was the uuid needed to belong to an offer that existed which was why we switched to doing it this way. Should I still cahnge it back?

Comment (tzumainn) [reply esi_leap/tests/db/sqlalchemy/test_api.py:292]: Ah, you're right - leave it for now!

**Comment by tzumainn** [comment]
Looks good - thanks!

**Pull request merged**